src/Quotient/UnsafeQuotient.idr

```diff
@@ -0,0 +1,32 @@
+module UnsafeQuotient
+
+import Quotient.Quotient
+
+%default total
+%access export
+
+private
+data InternalQuotientType : (x : Type) -> (eq : EqRel x) -> Type where
+  InternalWrap : x -> InternalQuotientType x eq
+
+QuotientType : (x : Type) -> (eq : EqRel x) -> Type
+QuotientType = InternalQuotientType
+
+Wrap : x -> QuotientType x eq
+Wrap = InternalWrap
+
+unwrap : QuotientType x eq -> x
+unwrap (InternalWrap a) = a
+
+wrapUnwrapId : (a : QuotientType x eq) -> a = Wrap (unwrap a)
+wrapUnwrapId (InternalWrap a) = Refl
+
+postulate
+QuotientEquality : (x : Type) -> (eq : EqRel x) -> (rel eq a b) -> Wrap a = Wrap b
+
+UnsafeQuotient : (x : Type) -> (eq : EqRel x) -> Quotient x eq
+UnsafeQuotient x eq = MkQuotient
+                        (QuotientType x eq)
+                        (Wrap ** (\a, b, h => QuotientEquality x eq h))
+                        (\y, f => ((\a => fst f $ unwrap a) ** (\a => Refl)))
+                        (\y, f, g, h, a => trans (cong $ wrapUnwrapId a) (sym $ h $ unwrap a))
```

Comment:
```suggestion
                        (\y, f, g, h, (InternalWrap a) => sym $ h a)
```

Reply:
I find these things very weird. In the previous line where I use `unwrap`, I cannot pattern match in the lambda, while in this line I can. The only reason why `unwrap` exists is because it didn't work there for me. Do you or @clayrat have any insight as to what might be the issue there?

Reply:
> As for the dependent pair, do you mean that it would be nicer if we didn't have to write `fst $ exists ...`? In that case, I'd just define another function.

Concretely, I suggest to replace the `exists` field with the following two fields (possibly with better chosen names):
```idris
  extension : (y : Type) -> (f : RespectingMap x y eq)
           -> carrier -> y
  commutes : (y : Type) -> (f : RespectingMap x y eq)
          -> extEq (fst f) (extension y f . (fst proj))
```
Why?
1. Most of the time, the function `extension y f` is probably what you want easy access to (what I tried to say above).
2. Should make it easier to build things stepwise using holes, because Idris doesn't need to juggle sigma types and first projections etc.
3. Might make it easier to talk about the code, e.g. 'the commuting triangle in `exists`' above is the `commutes` field.

> I'm not sure what you mean by 'the commuting triangle in `exists` would commute pointwise'.

What I meant was that in the `Quotients.Unsafe` module, the proof of `commutes y f a` is `Refl` for every `a`, which is nice, but outside this module, nothing reduces anymore.